elasticgit/storage/remote.py

```
@@ -0,0 +1,92 @@
+        path = [self.repo_name]
+        path.extend(parts)
+        return '%s://%s%s%s/%s.%s' % (
+            self.scheme,
+            self.host,
+            ':%s' % (self.port,) if self.port else '',
+            self.dir_name,
+            '/'.join(path),
+            self.suffix
+        )
+
+    def write_config(self, section, data):
+        raise RemoteStorageException(
+            'Remote storage is read only.')
+
+    def read_config(Self, section):
+        raise RemoteStorageException(
+            'Not implemented for remote storage.')
+
+    def storage_exists(self):
+        response = self.mk_request('GET', self.repo_url)
+        return response.ok
+
+    def destroy_storage(self):
+        raise RemoteStorageException(
+            'Remote storage is read only.')
+
+    def iterate(self, model_class):
+        response = self.mk_request('GET', self.url(fqcn(model_class)))
+        response.raise_for_status()
+        return [model_class(obj).set_read_only() for obj in response.json()]
+
+    def get(self, model_class, uuid):
+        response = self.mk_request('GET', self.url(fqcn(model_class), uuid))
+        response.raise_for_status()
+        return model_class(response.json()).set_read_only()
+
+    def store(self, model, message, author=None, committer=None):
+        raise RemoteStorageException(
+            'Remote storage is read only.')
+
+    def store_data(self, repo_path, data, message,
+                   author=None, committer=None):
+        raise RemoteStorageException(
+            'Remote storage is read only.')
+
+    def delete(self, model, message, author=None, committer=None):
+        raise RemoteStorageException(
+            'Remote storage is read only.')
+
+    def delete_data(self, repo_path, message,
+                    author=None, committer=None):
+        raise RemoteStorageException(
+            'Remote storage is read only.')
+
+    def pull(self, branch_name='master', remote_name='origin'):
+        pass
```

Should this maybe raise an exception instead? Otherwise it might seem like the remote repo has been updated while it hasn't.